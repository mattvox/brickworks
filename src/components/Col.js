import React from 'react';
import PropTypes from 'prop-types';

import getElementType from '../utils/getElementType';
import { gridStyles, itemStyles } from './styles';
import { itemTypes } from './types';
import wrapper from './hoc/wrapper';
import Layout from './Layout';

// prettier-ignore
export const Col = Layout.extend`
  ${itemStyles}

  ${({ grid }) => grid && gridStyles};
`;

const _Col = props => {
  const { as, children, ...rest } = props;
  const Element = Col.withComponent(getElementType(_Col, props));
  return (
    <Element className="brickworks-col" {...rest}>
      {children}
    </Element>
  );
};

export default wrapper(Col);

// const Wrapper = Component => props => {
//   const { as, children, ...rest } = props;
//   const Element = Component.withComponent(getElementType(Component, props));
//   return (
//     <Element className="brickworks-col" {...rest}>
//       {children}
//     </Element>
//   );
// };

// export default Wrapper(Col);

/* Additional props / will support? / function
* as - supported / renders el as another tag
* children - supported / allows passed children to be rendered

* className - should support / concat additional class names
* computer - should support / rename
* largeScreen - should support / rename
* mobile - should support / rename
* only - should support / only renders at specific breakpoint
* tablet - should support / rename
* textAlign - should support / sets text alignment of children
* widescreen - should support / rename

* floated - maybe / enums - left, right / already supported by flex/justify/align
* stretched - maybe / allows contents to take up entire grid height
* verticalAlign - maybe / renames align-items or align-content???
* width - maybe / can already do this with flex/flex-basis

* color - no support / allows row to be colored
*/

_Col.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node,
};

Col.propTypes = {
  ...itemTypes,
};

// export default _Col;
