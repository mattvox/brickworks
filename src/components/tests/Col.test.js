import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';

import testStyle from './helpers/testStyle';
import Col from '../Col';

Enzyme.configure({ adapter: new Adapter() });

describe('<Col />', () => {
  const wrapper = shallow(<Col />);

  it('should render correctly', () => {
    expect(wrapper.find('div').exists()).toBe(true);
  });

  describe('with default props/settings', () => {
    it('should have default style - flex: 0 1 auto', () => {
      expect(wrapper.find('div')).toHaveStyleRule('flex', '0 1 auto');
    });

    it('should match snapshot', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('with added props - 1st set', () => {
    const wrapperWithProps = shallow(
      <Col flex={[2, 3, '75%']} order={2} alignSelf="center" />,
    );

    const expectedStyleList = [
      { style: 'flex', value: '2 3 75%' },
      { style: 'order', value: '2' },
      { style: 'align-self', value: 'center' },
    ];

    it('should match snapshot with props', () => {
      expect(shallowToJson(wrapperWithProps)).toMatchSnapshot();
    });

    expectedStyleList.forEach(({ style, value }) => {
      return testStyle(wrapperWithProps, style, value);
    });
  });
});
