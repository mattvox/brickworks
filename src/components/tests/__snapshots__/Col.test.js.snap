// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> with added props - 1st set should match snapshot with props 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex: 2 3 75%;
  -ms-flex: 2 3 75%;
  flex: 2 3 75%;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

<div
  className="c0"
  order={2}
/>
`;

exports[`<Col /> with added props - 2nd set should match snapshot with props 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-box-flex: 5;
  -webkit-flex-grow: 5;
  -ms-flex-positive: 5;
  flex-grow: 5;
  -webkit-flex-shrink: 4;
  -ms-flex-negative: 4;
  flex-shrink: 4;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
}

<div
  className="c0"
/>
`;

exports[`<Col /> with default props/settings should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

<div
  className="c0"
/>
`;
